src/disco/get_dataset_listing.py

```diff
@@ -0,0 +1,29 @@
+import argparse
+import pandas as pd
+import json
+from pathlib import Path
+import math
+
+def main(args: argparse.Namespace):
+    """Main function to process DISCO data and output sample listing"""
+    # Read metadata
+    meta_df = pd.read_csv(args.metadata, sep="\t", dtype=str)
+    
+    # Process metadata
+    meta_dict = {
+        row['sample_id']: {k: v for k, v in {**row.to_dict(), 'sample_id': row['sample_id']}.items() 
```

Comment:
Add a `DISCO-` prefix (see TS2 portal) to all sample_ids for the listing. This will make it easier later to identify disco vs other portal IDs in other parts of the pipeline. (Note: the `DISCO-` prefix should be stripped when making the dataset_id as it should still be a resolvable IRI to the disco site).